tools/test/travis-ci/scancode-evaluate.py

```diff
@@ -0,0 +1,109 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations 
+"""
+
+# Asumptions for this script: 
+# 1. directory_name is scanned directory.
+#  Files are copied to this directory with full tree. As result, if we find
+#  license offender, we can have full path (just scrape directory_name). We do this
+#  magic because scancode allows to scan directories/one file.
+# 2. SPDX and license text is a must for all code files
+
+import json
+import argparse
+import sys
+import os.path
+import logging
+
+userlog = logging.getLogger("scancode-evaluate")
+userlog.setLevel(logging.INFO)
+
+MISSING_LICENSE_TEXT = "Missing license header"
+MISSING_PERMISIVE_LICENSE_TEXT = "Non-permissive license"
+MISSING_SPDX_TEXT = "Missing SPDX license identifier"
+
+FILE_OFFENDER_TEMPLATE = {
+   'file': None,
+   'reason': None
+}
+
+def license_check(directory_name, file):
+
+    offenders = []
+    try:
+        # find all licenses in the files, must be licensed and permissive
+        with open(file, 'r') as scancode_output:
+            results = json.load(scancode_output)
+    except ValueError:
+        userlog.warning("JSON could not be decoded")
+        return -1
+
+    try:
+        for file in results['files']:
+            license_offender = FILE_OFFENDER_TEMPLATE
+            license_offender['file'] = file
+            # ignore directory, not relevant here
+            if license_offender['file']['type'] == 'directory':
+                continue
+            if not license_offender['file']['licenses']:
+                license_offender['reason'] = MISSING_LICENSE_TEXT
+                offenders.append(license_offender)
+                continue
+
+            found_spdx = False
+            for i in range(len(license_offender['file']['licenses'])):
+                if license_offender['file']['licenses'][i]['category'] != 'Permissive':
+                    license_offender['reason'] = MISSING_PERMISIVE_LICENSE_TEXT
+                    offenders.append(license_offender)
+                # find SPDX, it shall be one of licenses found
+                if license_offender['file']['licenses'][i]['matched_rule']['identifier'].find("spdx") != -1:
+                    found_spdx = True
+
+            if not found_spdx:
+                license_offender['reason'] = MISSING_SPDX_TEXT
+                offenders.append(license_offender)
+
+    except KeyError:
+        userlog.warning("Invalid scancode json file")
+        return -1
+
+    if offenders:
+        userlog.warning("Found files with missing license details, please review and fix")
+        for offender in offenders:
+            userlog.warning("File: " + offender['file']['path'][len(directory_name):] + " " + "reason: " + offender['reason'])
```

Comment:
You can use C style string flags here ie "File: %s, %s", offender[][][], reason  ?

Reply:
I would leave it as it is